From ba79ba7871ee92f11880a305f90cce673bf189c9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20S=C5=82omi=C5=84ski?=
 <kacper.slominski72@gmail.com>
Date: Fri, 1 Nov 2024 01:59:08 +0100
Subject: [PATCH] Do not use the bundled Linux headers

---
 Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 1e76649..a511afc 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -380,7 +380,6 @@ EXTRA_DIST += \
 ###############################################################################
 
 default_includes = \
-	-I$(srcdir)/include/linux-private \
 	-I$(srcdir)/include \
 	-I$(builddir)/include \
 	$(NULL)
-- 
2.45.2

