From 00bb550096672af3cd01520f3f6544c4e674982e Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Thu, 27 Jun 2024 14:32:36 +0200
Subject: [PATCH] Add Managarm support

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 src/gui/openfilemanagerwindowjob_p.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/openfilemanagerwindowjob_p.h b/src/gui/openfilemanagerwindowjob_p.h
index 63ddcac..ca8f047 100644
--- a/src/gui/openfilemanagerwindowjob_p.h
+++ b/src/gui/openfilemanagerwindowjob_p.h
@@ -8,7 +8,7 @@
 #ifndef OPENFILEMANAGERWINDOWJOB_P_H
 #define OPENFILEMANAGERWINDOWJOB_P_H
 
-#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD)
+#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_MANAGARM)
 #define USE_DBUS 1
 #else
 #define USE_DBUS 0
-- 
2.45.2

