From 85652166ab3cee77b26f078d36f442561da7465b Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Wed, 14 Apr 2021 02:14:29 +0200
Subject: [PATCH 1/2] Remove calls to fsync

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 src/files.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/files.c b/src/files.c
index bc876cb1..efb719cb 100644
--- a/src/files.c
+++ b/src/files.c
@@ -1677,7 +1677,7 @@ bool make_backup_of(char *realname)
 
 	/* Since this backup is a newly created file, explicitly sync it to
 	 * permanent storage before starting to write out the actual file. */
-	if (fflush(backup_file) != 0 || fsync(fileno(backup_file)) != 0) {
+	if (fflush(backup_file) != 0) {// || fsync(fileno(backup_file)) != 0) {
 		fclose(backup_file);
 		goto problem;
 	}
@@ -1946,7 +1946,7 @@ bool write_file(const char *name, FILE *thefile, bool tmp,
 #endif
 
 	/* Ensure the data has reached the disk before reporting it as written. */
-	if (fflush(thefile) != 0 || fsync(fileno(thefile)) != 0) {
+	if (fflush(thefile) != 0) {// || fsync(fileno(thefile)) != 0) {
 		statusline(ALERT, _("Error writing %s: %s"), realname, strerror(errno));
 		fclose(thefile);
 		goto cleanup_and_exit;
-- 
2.31.0

