From 7209a9d3b8fce14afab12387bab65b08d9e882ce Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Sun, 28 Nov 2021 15:56:36 +0100
Subject: [PATCH] Remove a useless check in configure

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 configure.ac | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 23dd03c5..22063fdb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -71,12 +71,13 @@ AM_CONDITIONAL(BUILDING_FROM_GIT, test x$from_git = xyes)
 dnl Checks for pkg-config and gettext when building from git.
 
 if test x$from_git = xyes; then
-	if test ! -f $(aclocal --print-ac-dir)/pkg.m4; then
-		AC_MSG_ERROR([
-  *** The pkg.m4 macros are missing.
-  *** The pkg-config package needs to be installed when building from git.
-  *** After fixing this problem, rerun ./autogen.sh.])
-	fi
+dnl This test is currently broken on Managarm and unimportant
+dnl	if test ! -f $(aclocal --print-ac-dir)/pkg.m4; then
+dnl		AC_MSG_ERROR([
+dnl  *** The pkg.m4 macros are missing.
+dnl  *** The pkg-config package needs to be installed when building from git.
+dnl  *** After fixing this problem, rerun ./autogen.sh.])
+dnl	fi
 	if test "$ac_cv_path_MSGFMT" = ":"; then
 		AC_MSG_ERROR([
   *** The msgfmt program is missing.
-- 
2.34.0

