From 188d8ea3850074133889f2d00b082ccf89478881 Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Wed, 14 Jul 2021 03:10:06 +0200
Subject: [PATCH 1/2] Fill in the distribution specific anchor hook for Managarm

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 trust/trust-extract-compat | 16 ++++------------
 1 file changed, 4 insertions(+), 12 deletions(-)

diff --git a/trust/trust-extract-compat b/trust/trust-extract-compat
index b1b7a08..34051b3 100755
--- a/trust/trust-extract-compat
+++ b/trust/trust-extract-compat
@@ -17,16 +17,8 @@ if [ "$uid" != 0 ]; then
         exit 0
 fi
 
-echo "trust: the placeholder extract-compat command has not been customized by your distribution." >&2
+# Copy existing anchor modifications to /etc/ssl/local
+/usr/libexec/make-ca/copy-trust-modifications
 
-# You can use commands like this to extract data from trust modules
-# into appropriate locations for your distribution.
-#
-# trust extract --format=openssl-bundle --filter=ca-anchors \
-# 	--overwrite /tmp/openssl-bundle.pem
-# trust extract --format=pem-bundle --filter=ca-anchors --overwrite \
-# 	--purpose server-auth /tmp/server-auth-bundle.pem
-# trust extract --format=java-cacerts --filter=ca-anchors --overwrite \
-# 	--purpose server-auth /tmp/cacerts
-
-exit 1
+# Generate a new trust store
+/usr/sbin/make-ca -f -g
-- 
2.32.0

