From 9f16707dd860306cd501cb7ffdc11d7d27a4b8fe Mon Sep 17 00:00:00 2001
From: Matt Taylor <mstaveleytaylor@gmail.com>
Date: Fri, 7 May 2021 19:16:06 +0100
Subject: [PATCH] managarm: Initial port

---
 src/lib.rs | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/lib.rs b/src/lib.rs
index 6c8280f..8ae23f7 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -331,7 +331,8 @@ fn get_num_physical_cpus() -> usize {
     target_os = "android",
     target_os = "solaris",
     target_os = "illumos",
-    target_os = "fuchsia")
+    target_os = "fuchsia",
+    target_os = "managarm")
 )]
 fn get_num_cpus() -> usize {
     // On ARM targets, processors could be turned off to save power.
@@ -420,6 +421,7 @@ fn get_num_cpus() -> usize {
     target_os = "netbsd",
     target_os = "haiku",
     target_os = "hermit",
+    target_os = "managarm",
     windows,
 )))]
 fn get_num_cpus() -> usize {
-- 
2.31.1

