From d4b25740057a63b78de1987e13ae5a2cb60352b7 Mon Sep 17 00:00:00 2001
From: ikbenlike <thewatcheriswatchinu@gmail.com>
Date: Sat, 5 Aug 2023 03:09:26 +0200
Subject: [PATCH] remove glibc-specific feature test macros

---
 lib/fuse_lowlevel.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/lib/fuse_lowlevel.c b/lib/fuse_lowlevel.c
index fdef193..8a5b3bf 100644
--- a/lib/fuse_lowlevel.c
+++ b/lib/fuse_lowlevel.c
@@ -213,12 +213,8 @@ int fuse_send_reply_iov_nofree(fuse_req_t req, int error, struct iovec *iov,
 {
 	struct fuse_out_header out;
 
-#if __GLIBC__ >= 2 && __GLIBC_MINOR__ >= 32
 	const char *str = strerrordesc_np(error * -1);
 	if ((str == NULL && error != 0) || error > 0) {
-#else
-	if (error <= -1000 || error > 0) {
-#endif
 		fuse_log(FUSE_LOG_ERR, "fuse: bad error value: %i\n",	error);
 		error = -ERANGE;
 	}
-- 
2.34.1

