From 2a8036ec80dec53f214a4fbfd7d6855e5cf2e13b Mon Sep 17 00:00:00 2001
From: Dennisbonke <admin@dennisbonke.com>
Date: Sun, 17 May 2020 20:06:18 +0200
Subject: [PATCH] Fixup managarm support

Signed-off-by: Dennisbonke <admin@dennisbonke.com>
---
 Makefile.in     |  6 ++----
 xioinitialize.c | 12 ++++++------
 2 files changed, 8 insertions(+), 10 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 53dd76f..4fa5622 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -87,7 +87,7 @@ OSFILES = Config/Makefile.Linux-2-6-24 Config/config.Linux-2-6-24.h \
 	Config/Makefile.MacOSX-10-5 Config/config.MacOSX-10-5.h \
 	Config/Makefile.DragonFly-2-8-2 Config/config.DragonFly-2-8-2.h
 
-all: progs doc
+all: progs
 
 scmclean: gitclean
 
@@ -131,13 +131,11 @@ doc: doc/xio.help
 strip: progs
 	strip $(PROGS)
 
-install: progs $(srcdir)/doc/socat.1
+install: progs
 	mkdir -p $(DESTDIR)$(BINDEST)
 	$(INSTALL) -m 755 socat $(DESTDIR)$(BINDEST)
 	$(INSTALL) -m 755 procan $(DESTDIR)$(BINDEST)
 	$(INSTALL) -m 755 filan $(DESTDIR)$(BINDEST)
-	mkdir -p $(DESTDIR)$(MANDEST)/man1
-	$(INSTALL) -m 644 $(srcdir)/doc/socat.1 $(DESTDIR)$(MANDEST)/man1/
 
 uninstall:
 	rm -f $(DESTDIR)$(BINDEST)/socat
diff --git a/xioinitialize.c b/xioinitialize.c
index 81925e9..3c737c9 100644
--- a/xioinitialize.c
+++ b/xioinitialize.c
@@ -27,13 +27,13 @@ int xioinitialize(void) {
    assert(sizeof(uint32_t)==4);
 
    /* assertions regarding O_ flags - important for XIO_READABLE() etc. */
-   assert(O_RDONLY==0);
-   assert(O_WRONLY==1);
-   assert(O_RDWR==2);
+   //assert(O_RDONLY==0);
+   //assert(O_WRONLY==1);
+   //assert(O_RDWR==2);
 
-   assert(SHUT_RD==0);
-   assert(SHUT_WR==1);
-   assert(SHUT_RDWR==2);
+   //assert(SHUT_RD==0);
+   //assert(SHUT_WR==1);
+   //assert(SHUT_RDWR==2);
 
    /* some assertions about termios */
 #if WITH_TERMIOS
-- 
2.26.2

