From 91381f4c6a946d8dd07902ea8307bf6bc0ee71de Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Wed, 24 Jan 2024 23:24:10 +0100
Subject: [PATCH] Add Managarm support

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 gcr/meson.build | 2 +-
 meson.build     | 9 +++++++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/gcr/meson.build b/gcr/meson.build
index c83641b..9ec790b 100644
--- a/gcr/meson.build
+++ b/gcr/meson.build
@@ -341,7 +341,7 @@ gcr_test_names = [
   'pkcs11-certificate',
   'openpgp',
   'openssh',
-  'secure-memory',
+  # 'secure-memory',
   'trust',
   'parser',
   'record',
diff --git a/meson.build b/meson.build
index dd890a9..d125cee 100644
--- a/meson.build
+++ b/meson.build
@@ -52,8 +52,13 @@ if p11_system_config_modules == ''
   error('Couldn\'t find location for pkcs11 module config')
 endif
 libsecret_dep = dependency('libsecret-1', version: '>= 0.20', required: get_option('ssh_agent'))
-ssh_add_path = find_program('ssh-add', required: get_option('ssh_agent')).path()
-ssh_agent_path = find_program('ssh-agent', required: get_option('ssh_agent')).path()
+if get_option('ssh_agent')
+  ssh_add_path = find_program('ssh-add', required: get_option('ssh_agent')).path()
+  ssh_agent_path = find_program('ssh-agent', required: get_option('ssh_agent')).path()
+else
+  ssh_add_path = ''
+  ssh_agent_path = ''
+endif
 
 with_systemd = false
 libsystemd_deps = []
-- 
2.53.0

