From c939d215dfd398f77410fb92041024b4b9097f0b Mon Sep 17 00:00:00 2001
From: no92 <no92.mail@gmail.com>
Date: Fri, 22 Jan 2021 18:33:00 +0100
Subject: [PATCH libevdev 1/2] Add stddef.h for missing size_t

---
 libevdev/libevdev.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libevdev/libevdev.h b/libevdev/libevdev.h
index 30e4d6f..4f5f4b6 100644
--- a/libevdev/libevdev.h
+++ b/libevdev/libevdev.h
@@ -32,6 +32,7 @@ extern "C" {
 
 #include <linux/input.h>
 #include <stdarg.h>
+#include <stddef.h>
 
 #define LIBEVDEV_ATTRIBUTE_PRINTF(_format, _args) __attribute__ ((format (printf, _format, _args)))
 
-- 
2.34.1

