From 787a0187178ee286a671d0ef8d5b352bf2d06037 Mon Sep 17 00:00:00 2001
From: no92 <no92.mail@gmail.com>
Date: Wed, 21 Aug 2024 00:16:01 +0200
Subject: [PATCH] meta: add meson.build

---
 meson.build | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)
 create mode 100644 meson.build

diff --git a/meson.build b/meson.build
new file mode 100644
index 0000000..02eb366
--- /dev/null
+++ b/meson.build
@@ -0,0 +1,23 @@
+project('freebsd-e1000', ['c'])
+
+inc = include_directories('.')
+
+src = files(
+	'e1000_mbx.c',
+	'e1000_82542.c',
+	'e1000_phy.c',
+	'e1000_82543.c',
+	'e1000_ich8lan.c',
+	'e1000_82575.c',
+	'e1000_82540.c',
+	'e1000_i210.c',
+	'e1000_82541.c',
+	'e1000_api.c',
+	'e1000_manage.c',
+	'e1000_vf.c',
+	'e1000_mac.c',
+	'e1000_82571.c',
+	'e1000_base.c',
+	'e1000_nvm.c',
+	'e1000_80003es2lan.c',
+)
-- 
2.45.1

