From 7bd5cf6b0877a6e52fac9d0e43962b8f2c73f3e7 Mon Sep 17 00:00:00 2001
From: Matt Taylor <mstaveleytaylor@gmail.com>
Date: Thu, 27 May 2021 20:10:44 +0100
Subject: [PATCH] Support managarm OS

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 neofetch | 43 ++++++++++++++++++++++++++++++++++++-------
 1 file changed, 36 insertions(+), 7 deletions(-)

diff --git a/neofetch b/neofetch
index 62b82ea..bd0db18 100755
--- a/neofetch
+++ b/neofetch
@@ -62,17 +62,17 @@ print_info() {
     info "Uptime" uptime
     info "Packages" packages
     info "Shell" shell
-    info "Resolution" resolution
+    # info "Resolution" resolution
     info "DE" de
     info "WM" wm
     info "WM Theme" wm_theme
     info "Theme" theme
     info "Icons" icons
-    info "Terminal" term
-    info "Terminal Font" term_font
-    info "CPU" cpu
-    info "GPU" gpu
-    info "Memory" memory
+    # info "Terminal" term
+    # info "Terminal Font" term_font
+    # info "CPU" cpu
+    # info "GPU" gpu
+    # info "Memory" memory
 
     # info "GPU Driver" gpu_driver  # Linux/macOS only
     # info "Disk" disk
@@ -945,6 +945,7 @@ get_os() {
         AIX)      os=AIX ;;
         IRIX*)    os=IRIX ;;
         FreeMiNT) os=FreeMiNT ;;
+        managarm) os=Managarm ;;
 
         Linux|GNU*)
             os=Linux
@@ -1205,6 +1206,10 @@ get_distro() {
         FreeMiNT)
             distro=FreeMiNT
         ;;
+
+        Managarm)
+            distro=Managarm
+        ;;
     esac
 
     distro=${distro//Enterprise Server}
@@ -1339,6 +1344,10 @@ get_model() {
             model=$(sysctl -n hw.model)
             model=${model/ (_MCH *)}
         ;;
+
+        Managarm)
+            model="Unknown model"
+        ;;
     esac
 
     # Remove dummy OEM info.
@@ -8136,6 +8145,26 @@ o00.              k0O${c2}dddddd${c1}occ
 EOF
         ;;
 
+        "Managarm"*)
+            # TODO: Add proper art
+            set_colors 2 1 3 5
+            read -rd '' ascii_data <<'EOF'
+${c1}                   @@. @@                         
+                   %%%@@%@                        
+                 &%%@%&%%%.                       
+                @%%@@%%%%%@%@@@@/                 
+               %%%@@@%%%%%%%%%%%%%%@              
+          #@%%%%%%%%%%%%%%@@@@@@@@%%%@            
+      @%%%%%%%%%%%%%%%%%@@@@@@@@&%%%@@@%%%&@@     
+    @@%@@%%%%%%%%%%%%%%@@@@@@@@@@@@@@@@@@@@@@@%%%@
+   @%%%%%%%%%%%%%%%%%%%@@@@@@@@@@@@@@@@@@@@@@@@%@ 
+  @  @%%%%%%%%%%%%%%%%%%@@@@@@@@@@@@@@@@@@@@@%%@  
+  @%%%%%%%%%%%%%%%%%%%%%@@@@@@@@@@@%%%%%%%%%@%    
+ @. @%%%%%%%%%%%%%%%%%%%%%%%%%%%%%@@@.            
+    @%%%%%%&@@@@(///#@@@%,
+EOF
+        ;;
+
         "Mandriva"*)
             set_colors 4 3
             read -rd '' ascii_data <<'EOF'
@@ -10913,7 +10942,7 @@ main() {
     eval "$config"
 
     get_args "$@"
-    [[ $verbose != on ]] && exec 2>/dev/null
+    # [[ $verbose != on ]] && exec 2>/dev/null
     get_simple "$@"
     get_distro
     get_bold
-- 
2.42.0

