From 58f93d7da02601cce17e771b5f9126507374fd60 Mon Sep 17 00:00:00 2001
From: no92 <no92.mail@gmail.com>
Date: Sat, 29 Mar 2025 12:32:14 +0100
Subject: [PATCH] Add mlibc support

---
 src/unix.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/unix.rs b/src/unix.rs
index 5536eef..42e56bc 100644
--- a/src/unix.rs
+++ b/src/unix.rs
@@ -95,6 +95,7 @@ extern "C" {
             target_os = "redox",
             target_os = "dragonfly",
             target_os = "emscripten",
+            target_env = "mlibc",
         ),
         link_name = "__errno_location"
     )]
-- 
2.54.0

