From 4e2b6158920e51632d650784388ca4550a49e10f Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Sun, 16 Apr 2023 20:35:30 +0200
Subject: [PATCH 2/3] Enable DMABUF for managarm

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 src/gallium/auxiliary/util/u_screen.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gallium/auxiliary/util/u_screen.c b/src/gallium/auxiliary/util/u_screen.c
index 348aaa8..a1e517f 100644
--- a/src/gallium/auxiliary/util/u_screen.c
+++ b/src/gallium/auxiliary/util/u_screen.c
@@ -442,7 +442,7 @@ u_pipe_screen_get_param_defaults(struct pipe_screen *pscreen,
       return 0;
 
    case PIPE_CAP_DMABUF:
-#if defined(HAVE_LIBDRM) && (DETECT_OS_LINUX || DETECT_OS_BSD)
+#if defined(HAVE_LIBDRM) && (DETECT_OS_LINUX || DETECT_OS_BSD || DETECT_OS_MANAGARM)
       fd = pscreen->get_screen_fd(pscreen);
       if (fd != -1 && (drmGetCap(fd, DRM_CAP_PRIME, &cap) == 0))
          return cap;
-- 
2.42.0

