From 2e33718d41362fd893aee44a4200fbb38ae99253 Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Sun, 16 Apr 2023 20:35:30 +0200
Subject: [PATCH 2/3] Enable DMABUF for managarm

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 src/gallium/auxiliary/util/u_screen.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gallium/auxiliary/util/u_screen.c b/src/gallium/auxiliary/util/u_screen.c
index f0e1b7e..b6a1622 100644
--- a/src/gallium/auxiliary/util/u_screen.c
+++ b/src/gallium/auxiliary/util/u_screen.c
@@ -423,7 +423,7 @@ u_pipe_screen_get_param_defaults(struct pipe_screen *pscreen,
       return 0;
 
    case PIPE_CAP_DMABUF:
-#if DETECT_OS_LINUX || DETECT_OS_BSD
+#if DETECT_OS_LINUX || DETECT_OS_BSD || DETECT_OS_MANAGARM
       return 1;
 #else
       return 0;
-- 
2.40.0

