From 4d30ae79fb88194ed1d1e9d4efc052c0a92f4d9c Mon Sep 17 00:00:00 2001
From: Alexander van der Grinten <alexander.vandergrinten@gmail.com>
Date: Wed, 18 Dec 2019 21:14:21 +0100
Subject: [PATCH 2/2] Work around cross compilation issue

---
 meson.build | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index cbf7aab..55a9a0a 100644
--- a/meson.build
+++ b/meson.build
@@ -1550,8 +1550,9 @@ endif
 # TODO: symbol mangling
 
 if with_platform_wayland
-  dep_wl_scanner = dependency('wayland-scanner', native: true)
-  prog_wl_scanner = find_program(dep_wl_scanner.get_pkgconfig_variable('wayland_scanner'))
+  # We assume that we find the same version of wayland-scanner on build system and target.
+  dep_wl_scanner = dependency('wayland-scanner')
+  prog_wl_scanner = find_program('wayland-scanner')
   if dep_wl_scanner.version().version_compare('>= 1.15')
     wl_scanner_arg = 'private-code'
   else
-- 
2.20.1

