From 67b7d1ef8fbb9b3878a178f3244e102c7a4bd552 Mon Sep 17 00:00:00 2001
From: Alexander van der Grinten <alexander.vandergrinten@gmail.com>
Date: Wed, 18 Dec 2019 21:14:21 +0100
Subject: [PATCH 3/3] Work around cross compilation issue

---
 meson.build | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 91d9796..3facf62 100644
--- a/meson.build
+++ b/meson.build
@@ -1313,8 +1313,9 @@ endif
 # TODO: symbol mangling
 
 if with_platform_wayland
-  dep_wl_scanner = dependency('wayland-scanner', native: true)
-  prog_wl_scanner = find_program(dep_wl_scanner.get_pkgconfig_variable('wayland_scanner'))
+  # We assume that we find the same version of wayland-scanner on build system and target.
+  dep_wl_scanner = dependency('wayland-scanner')
+  prog_wl_scanner = find_program('wayland-scanner')
   if dep_wl_scanner.version().version_compare('>= 1.15')
     wl_scanner_arg = 'private-code'
   else
-- 
2.20.1

