From 24165e8bef23a301ac4757f11db73a6f9a63df0a Mon Sep 17 00:00:00 2001
From: no92 <no92.mail@gmail.com>
Date: Mon, 26 Feb 2024 21:06:41 +0100
Subject: [PATCH 4/4] Define _GNU_SOURCE for managarm

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 2c7691d..f3027cb 100644
--- a/meson.build
+++ b/meson.build
@@ -909,7 +909,7 @@ if cc.has_function('fmemopen')
 endif
 
 # TODO: this is very incomplete
-if ['linux', 'cygwin', 'gnu', 'freebsd', 'gnu/kfreebsd', 'haiku', 'android'].contains(host_machine.system())
+if ['linux', 'cygwin', 'gnu', 'freebsd', 'gnu/kfreebsd', 'haiku', 'android', 'managarm'].contains(host_machine.system())
   pre_args += '-D_GNU_SOURCE'
 elif host_machine.system() == 'sunos'
   pre_args += '-D__EXTENSIONS__'
-- 
2.43.2

