From a9f880255e400d482e27714307d38feb64ef5b18 Mon Sep 17 00:00:00 2001
From: no92 <no92.mail@gmail.com>
Date: Sat, 23 Jan 2021 21:09:10 +0100
Subject: [PATCH 3/3] Even more managarm-specific changes

---
 xf86drm.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/xf86drm.c b/xf86drm.c
index 3ab71f7..26fb8bc 100644
--- a/xf86drm.c
+++ b/xf86drm.c
@@ -3314,7 +3314,7 @@ drm_public char *drmGetDeviceNameFromFd(int fd)
 
 static bool drmNodeIsDRM(int maj, int min)
 {
-#ifdef __linux__
+#if defined(__linux__) || defined(__managarm__)
     char path[64];
     struct stat sbuf;
 
@@ -3571,7 +3571,7 @@ sysfs_uevent_get(const char *path, const char *fmt, ...)
 /* Little white lie to avoid major rework of the existing code */
 #define DRM_BUS_VIRTIO 0x10
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__managarm__)
 static int get_subsystem_type(const char *device_path)
 {
     char path[PATH_MAX + 1] = "";
@@ -3637,7 +3637,7 @@ static int drmParseSubsystemType(int maj, int min)
 #endif
 }
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__managarm__)
 static void
 get_pci_path(int maj, int min, char *pci_path)
 {
@@ -4121,7 +4121,7 @@ free_device:
     return ret;
 }
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__managarm__)
 static int drm_usb_dev_path(int maj, int min, char *path, size_t len)
 {
     char *value, *tmp_path, *slash;
@@ -4279,7 +4279,7 @@ free_device:
 
 static int drmParseOFBusInfo(int maj, int min, char *fullname)
 {
-#ifdef __linux__
+#if defined(__linux__) || defined(__managarm__)
     char path[PATH_MAX + 1], *name, *tmp_name;
 
     snprintf(path, sizeof(path), "/sys/dev/char/%d:%d/device", maj, min);
@@ -4314,7 +4314,7 @@ static int drmParseOFBusInfo(int maj, int min, char *fullname)
 
 static int drmParseOFDeviceInfo(int maj, int min, char ***compatible)
 {
-#ifdef __linux__
+#if defined(__linux__) || defined(__managarm__)
     char path[PATH_MAX + 1], *value, *tmp_name;
     unsigned int count, i;
     int err;
-- 
2.39.5

