From 89c6c30625b487160f162212c20ed54ff996b6b6 Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Fri, 29 Apr 2022 21:47:31 +0200
Subject: [PATCH 3/3] Remove docs calls

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 autogen.sh | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/autogen.sh b/autogen.sh
index 6115a87..b1defea 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -1,6 +1,3 @@
 #!/bin/sh
 
 autoreconf -f -i -s
-
-cd man
-./make.sh
-- 
2.36.0

