From e54d5cb8f5a11278f1e8b7e005e5c5573a4e08b5 Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Fri, 29 Apr 2022 21:47:31 +0200
Subject: [PATCH 3/4] Remove docs calls

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 autogen.sh | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/autogen.sh b/autogen.sh
index 37feedd..cd10c4c 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -1,6 +1,3 @@
 #!/bin/sh
 set -xe
 autoreconf -f -i -s
-
-cd man
-./make.sh
-- 
2.47.0

