From 61b269aae765bfa3d11c679b90b2ac7c7806cfa3 Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Mon, 13 Oct 2025 22:58:02 +0200
Subject: [PATCH 2/2] Add support for GI_LDD_WRAPPER environment variable

---
 giscanner/shlibs.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/giscanner/shlibs.py b/giscanner/shlibs.py
index 66cbb25..a004f34 100644
--- a/giscanner/shlibs.py
+++ b/giscanner/shlibs.py
@@ -97,9 +97,12 @@ def _resolve_non_libtool(options, binary, libraries):
             args.extend(libtool)
             args.append('--mode=execute')
         platform_system = platform.system()
+        ldd_wrapper_from_env = os.environ.get("GI_LDD_WRAPPER")
         if options.ldd_wrapper:
             args.extend(options.ldd_wrapper)
             args.append(binary.args[0])
+        elif ldd_wrapper_from_env:
+            args.extend([ldd_wrapper_from_env, binary.args[0]])
         elif platform_system == 'Darwin':
             args.extend(['otool', '-L', binary.args[0]])
         else:
-- 
2.51.0

