From 717a54091fe9e2b71d5e00c44b776d8c2c33b0b1 Mon Sep 17 00:00:00 2001
From: Dennisbonke <admin@dennisbonke.com>
Date: Sun, 26 Jul 2020 00:19:58 +0200
Subject: [PATCH] Add xwayland support for managarm

Signed-off-by: Dennisbonke <admin@dennisbonke.com>
---
 compositor/xwayland.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/compositor/xwayland.c b/compositor/xwayland.c
index 9881cd9c..0b133f14 100644
--- a/compositor/xwayland.c
+++ b/compositor/xwayland.c
@@ -125,10 +125,11 @@ spawn_xserver(void *user_data, const char *display, int abstract_fd, int unix_fd
 			  "-listen", unix_fd_str,
 			  "-wm", wm_fd_str,
 			  "-terminate",
+			  "-ac",
 			  NULL) < 0)
 			weston_log("exec of '%s %s -rootless "
 				   "-listen %s -listen %s -wm %s "
-				   "-terminate' failed: %m\n",
+				   "-terminate -ac' failed: %m\n",
 				   xserver, display,
 				   abstract_fd_str, unix_fd_str, wm_fd_str);
 	fail:
-- 
2.28.0.rc2

