From 3665504bec574129b4f8b3d86c3a3cb982c7d7b1 Mon Sep 17 00:00:00 2001
From: Matt Taylor <mstaveleytaylor@gmail.com>
Date: Mon, 14 Feb 2022 01:31:01 +0000
Subject: [PATCH 1/2] managarm: use dl_iterate_phdr

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 src/symbolize/gimli.rs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/symbolize/gimli.rs b/src/symbolize/gimli.rs
index 31b83b5..a79d17f 100644
--- a/src/symbolize/gimli.rs
+++ b/src/symbolize/gimli.rs
@@ -43,6 +43,7 @@ cfg_if::cfg_if! {
         target_os = "solaris",
         target_os = "illumos",
         target_os = "aix",
+        target_os = "managarm",
     ))] {
         #[path = "gimli/mmap_unix.rs"]
         mod mmap;
@@ -228,6 +229,7 @@ cfg_if::cfg_if! {
             target_os = "netbsd",
             target_os = "nto",
             target_os = "android",
+            target_os = "managarm",
         ),
         not(target_env = "uclibc"),
     ))] {
-- 
2.51.0

