From bfc377608dad078fd25d569cbf5125d71296ef24 Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Sat, 18 Oct 2025 16:08:37 +0200
Subject: [PATCH] Add Managarm support

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 gdk/loaders/gdkpng.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/gdk/loaders/gdkpng.c b/gdk/loaders/gdkpng.c
index b82fd05..f642100 100644
--- a/gdk/loaders/gdkpng.c
+++ b/gdk/loaders/gdkpng.c
@@ -344,7 +344,11 @@ gdk_load_png (GBytes      *bytes,
   png_set_read_user_chunk_fn (png, &cicp, png_read_chunk_func);
 #endif
 
+#ifdef __mlibc__
+  if (setjmp (png_jmpbuf (png)))
+#else
   if (sigsetjmp (png_jmpbuf (png), 1))
+#endif
     {
       g_free (buffer);
       g_free (row_pointers);
@@ -639,7 +643,11 @@ gdk_save_png (GdkTexture *texture)
   gdk_color_state_ref (color_state);
   bytes = NULL;
 
+#ifdef __mlibc__
+  if (setjmp (png_jmpbuf (png)))
+#else
   if (sigsetjmp (png_jmpbuf (png), 1))
+#endif
     {
       gdk_color_state_unref (color_state);
       g_clear_pointer (&bytes, g_bytes_unref);
-- 
2.51.0

