From 361dcf44a22388d3d0985a089cdc5cc47f939933 Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Tue, 4 Oct 2022 02:30:22 +0200
Subject: [PATCH 2/2] Fix lt_sysroot

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 build-aux/ltmain.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build-aux/ltmain.in b/build-aux/ltmain.in
index a5f21a1..c06d55c 100644
--- a/build-aux/ltmain.in
+++ b/build-aux/ltmain.in
@@ -6497,7 +6497,7 @@ func_mode_link ()
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir=-L$libdir
+	      add_dir=-L$lt_sysroot$libdir
 	      # Try looking first in the location we're being installed to.
 	      if test -n "$inst_prefix_dir"; then
 		case $libdir in
-- 
2.37.2

