From 9cf1d5ca48c9da110879ee8d7c95ba3303149204 Mon Sep 17 00:00:00 2001
From: avdgrinten <alexander.vandergrinten@gmail.com>
Date: Sat, 29 Oct 2016 12:21:16 +0200
Subject: [PATCH 2/2] Add managarm target

---
 lib/config.sub | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/config.sub b/lib/config.sub
index c894da4..c89ec11 100755
--- a/lib/config.sub
+++ b/lib/config.sub
@@ -1362,7 +1362,8 @@ case $os in
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
+		  | -managarm*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
-- 
2.11.0

