From ca383ca13d15870d35cb5e698b5c5485b30e9506 Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Tue, 13 Apr 2021 02:07:40 +0200
Subject: [PATCH] Add missing include

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 lib/curl_setup.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/curl_setup.h b/lib/curl_setup.h
index cf1c26a14..579945e5d 100644
--- a/lib/curl_setup.h
+++ b/lib/curl_setup.h
@@ -160,6 +160,7 @@
 /* ================================================================ */
 
 #include <curl/curl.h>
+#include <sys/select.h>
 
 /*
  * Disable other protocols when http is the only one desired.
-- 
2.31.0

