From d926ae4df05650a7af615ae92cb98493cc7409fb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20S=C5=82omi=C5=84ski?=
 <kacper.slominski72@gmail.com>
Date: Tue, 21 Dec 2021 13:07:19 +0100
Subject: [PATCH 1/2] configure: add managarm support

---
 configure | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/configure b/configure
index 48c2177..a6e9161 100755
--- a/configure
+++ b/configure
@@ -521,6 +521,8 @@ elif check_define __NetBSD__; then
   targetos='NetBSD'
 elif check_define __APPLE__; then
   targetos='Darwin'
+elif check_define __managarm__; then
+  targetos='Managarm'
 else
   # This is a fatal error, but don't report it yet, because we
   # might be going to just print the --help text, or it might
@@ -677,6 +679,9 @@ Linux)
   linux_user="yes"
   vhost_user=${default_feature:-yes}
 ;;
+Managarm)
+  managarm="yes"
+;;
 esac
 
 : ${make=${MAKE-make}}
@@ -3944,6 +3949,9 @@ if test "$skip_meson" = no; then
     if test "$darwin" = "yes" ; then
         echo "system = 'darwin'" >> $cross
     fi
+    if test "$managarm" = "yes" ; then
+        echo "system = 'managarm'" >> $cross
+    fi
     case "$ARCH" in
         i386)
             echo "cpu_family = 'x86'" >> $cross
-- 
2.34.1

