From 42800eff3993ac6377b8abc884e668ecfbe46fc9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20S=C5=82omi=C5=84ski?=
 <kacper.slominski72@gmail.com>
Date: Tue, 21 Dec 2021 13:07:19 +0100
Subject: [PATCH 1/6] configure: add managarm support

---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index 1833637..78b077a 100755
--- a/configure
+++ b/configure
@@ -360,6 +360,8 @@ elif check_define __NetBSD__; then
   host_os=netbsd
 elif check_define __APPLE__; then
   host_os=darwin
+elif check_define __managarm__; then
+  host_os=managarm
 else
   # This is a fatal error, but don't report it yet, because we
   # might be going to just print the --help text, or it might
-- 
2.49.0

