From 520703e64241fe22153393eebe1ae9766fa7f8c7 Mon Sep 17 00:00:00 2001
From: no92 <no92.mail@gmail.com>
Date: Wed, 24 Jan 2024 19:34:27 +0100
Subject: [PATCH 3/5] Add managarm support

---
 block/file-posix.c | 11 +++++++++--
 meson.build        |  1 +
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/block/file-posix.c b/block/file-posix.c
index ff928b5..cf32e77 100644
--- a/block/file-posix.c
+++ b/block/file-posix.c
@@ -110,6 +110,13 @@
 #include <sys/diskslice.h>
 #endif
 
+#ifdef __managarm__
+#include <sys/ioctl.h>
+#if defined(CONFIG_BLKZONED)
+#include <linux/blkzoned.h>
+#endif
+#endif
+
 /* OS X does not have O_DSYNC */
 #ifndef O_DSYNC
 #ifdef O_SYNC
@@ -1217,7 +1224,7 @@ static int hdev_get_max_hw_transfer(int fd, struct stat *st)
 /*
  * Get a sysfs attribute value as character string.
  */
-#ifdef CONFIG_LINUX
+#if defined(CONFIG_LINUX) || defined(CONFIG_MANAGARM)
 static int get_sysfs_str_val(struct stat *st, const char *attribute,
                              char **val) {
     g_autofree char *sysfspath = NULL;
@@ -1271,7 +1278,7 @@ static int get_sysfs_zoned_model(struct stat *st, BlockZoneModel *zoned)
 /*
  * Get a sysfs attribute value as a long integer.
  */
-#ifdef CONFIG_LINUX
+#if defined(CONFIG_LINUX) || defined(CONFIG_MANAGARM)
 static long get_sysfs_long_val(struct stat *st, const char *attribute)
 {
     g_autofree char *str = NULL;
diff --git a/meson.build b/meson.build
index a11018b..ab40bf0 100644
--- a/meson.build
+++ b/meson.build
@@ -2370,6 +2370,7 @@ config_host_data.set('CONFIG_PIXMAN', pixman.found())
 config_host_data.set('CONFIG_SLIRP', slirp.found())
 config_host_data.set('CONFIG_SNAPPY', snappy.found())
 config_host_data.set('CONFIG_SOLARIS', host_os == 'sunos')
+config_host_data.set('CONFIG_MANAGARM', host_os == 'managarm')
 if get_option('tcg').allowed()
   config_host_data.set('CONFIG_TCG', 1)
   config_host_data.set('CONFIG_TCG_INTERPRETER', tcg_arch == 'tci')
-- 
2.47.0

