From 3dcb11b0eb65260f9c4f5be14518ebe374f64d4f Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Wed, 2 Aug 2023 04:45:55 +0200
Subject: [PATCH 3/7] managarm: Add missing glue for memmap, num_cpus and users

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 src/unix/mlibc/mod.rs | 59 ++++++++++++++++++++++++++-----------------
 1 file changed, 36 insertions(+), 23 deletions(-)

diff --git a/src/unix/mlibc/mod.rs b/src/unix/mlibc/mod.rs
index e7c4cc7..eb8cdc6 100644
--- a/src/unix/mlibc/mod.rs
+++ b/src/unix/mlibc/mod.rs
@@ -130,8 +130,8 @@ s! {
         pub si_signo: ::c_int,
         pub si_code: ::c_int,
         pub si_errno: ::c_int,
-        pub si_pid: pid_t,
-        pub si_uid: uid_t,
+        pub si_pid: ::pid_t,
+        pub si_uid: ::uid_t,
         pub si_addr: *mut ::c_void,
         pub si_status: ::c_int,
         pub si_value: sigval,
@@ -188,9 +188,6 @@ pub type blkcnt_t = ::c_long;
 // abis/mlibc/nlink_t.h
 pub type nlink_t = ::c_int;
 
-// abis/mlibc/pid_t.h
-pub type pid_t = ::c_int;
-
 // options/posix/include/bits/posix/in_addr_t.h
 pub type in_addr_t = u32;
 
@@ -217,6 +214,7 @@ pub const STDERR_FILENO: ::c_int = 2;
 pub const STDIN_FILENO: ::c_int = 0;
 pub const STDOUT_FILENO: ::c_int = 1;
 pub const _SC_GETPW_R_SIZE_MAX: ::c_int = 1;
+pub const _SC_NPROCESSORS_ONLN: ::c_int = 6;
 pub const _SC_PAGESIZE: ::c_int = _SC_PAGE_SIZE;
 pub const _SC_PAGE_SIZE: ::c_int = 3;
 
@@ -347,24 +345,36 @@ pub const S_IFLNK: mode_t = 0xA000;
 pub const S_IFMT: mode_t = 0xF000;
 pub const S_IFREG: mode_t = 0x8000;
 pub const S_IFSOCK: mode_t = 0xC000;
+pub const S_IRGRP: mode_t = 0o40;
+pub const S_IROTH: mode_t = 0o4;
+pub const S_IRUSR: mode_t = 0o400;
+pub const S_IRWXG: mode_t = 0o70;
+pub const S_IRWXO: mode_t = 0o7;
+pub const S_IRWXU: mode_t = 0o700;
+pub const S_IWGRP: mode_t = 0o20;
+pub const S_IWOTH: mode_t = 0o2;
+pub const S_IWUSR: mode_t = 0o200;
+pub const S_IXGRP: mode_t = 0o10;
+pub const S_IXOTH: mode_t = 0o1;
+pub const S_IXUSR: mode_t = 0o100;
 s! {
     pub struct stat {
-        pub st_dev: dev_t,
-        pub st_ino: ino_t,
-        pub st_mode: mode_t,
-        pub st_nlink: nlink_t,
-        pub st_uid: uid_t,
-        pub st_gid: gid_t,
-        pub st_rdev: dev_t,
-        pub st_size: off_t,
-        pub st_atime: time_t,
-        pub st_atime_nsec: c_long,
-        pub st_mtime: time_t,
-        pub st_mtime_nsec: c_long,
-        pub st_ctime: time_t,
-        pub st_ctime_nsec: c_long,
-        pub st_blksize: blksize_t,
-        pub st_blocks: blkcnt_t,
+        pub st_dev: ::dev_t,
+        pub st_ino: ::ino_t,
+        pub st_mode: ::mode_t,
+        pub st_nlink: ::nlink_t,
+        pub st_uid: ::uid_t,
+        pub st_gid: ::gid_t,
+        pub st_rdev: ::dev_t,
+        pub st_size: ::off_t,
+        pub st_atime: ::time_t,
+        pub st_atime_nsec: ::c_long,
+        pub st_mtime: ::time_t,
+        pub st_mtime_nsec: ::c_long,
+        pub st_ctime: ::time_t,
+        pub st_ctime_nsec: ::c_long,
+        pub st_blksize: ::blksize_t,
+        pub st_blocks: ::blkcnt_t,
     }
 }
 
@@ -448,8 +458,8 @@ s! {
     pub struct passwd {
         pub pw_name: *mut ::c_char,
         pub pw_passwd: *mut ::c_char,
-        pub pw_uid: uid_t,
-        pub pw_gid: gid_t,
+        pub pw_uid: ::uid_t,
+        pub pw_gid: ::gid_t,
         pub pw_gecos: *mut ::c_char,
         pub pw_dir: *mut ::c_char,
         pub pw_shell: *mut ::c_char,
@@ -699,6 +709,8 @@ extern "C" {
         result: *mut *mut passwd,
     ) -> ::c_int;
     pub fn ioctl(fd: ::c_int, request: ::c_ulong, ...) -> ::c_int;
+    pub fn mprotect(addr: *mut ::c_void, len: ::size_t, prot: ::c_int) -> ::c_int;
+    pub fn msync(addr: *mut ::c_void, len: ::size_t, flags: ::c_int) -> ::c_int;
     pub fn pthread_condattr_setclock(
         attr: *mut pthread_condattr_t,
         clock_id: ::clockid_t,
@@ -721,5 +733,6 @@ extern "C" {
         addrlen: *mut ::socklen_t,
     ) -> ::ssize_t;
     pub fn setgroups(ngroups: ::c_int, ptr: *const ::gid_t) -> ::c_int;
+    pub fn setpwent();
     pub fn writev(fd: ::c_int, iov: *const ::iovec, count: ::c_int) -> ::ssize_t;
 }
-- 
2.42.0

