From baf9678b41e461cfdbce9cd3906966ca9b6244fd Mon Sep 17 00:00:00 2001
From: Matt Taylor <mstaveleytaylor@gmail.com>
Date: Wed, 16 Feb 2022 17:45:17 +0000
Subject: [PATCH 4/4] managarm: add missing constants for 1.58.1

---
 src/unix/mlibc/mod.rs | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/unix/mlibc/mod.rs b/src/unix/mlibc/mod.rs
index cd7e4dd..54de74f 100644
--- a/src/unix/mlibc/mod.rs
+++ b/src/unix/mlibc/mod.rs
@@ -311,6 +311,7 @@ pub const EXDEV: ::c_int = 1075;
 
 // options/posix/include/fcntl.h
 pub const AT_FDCWD: ::c_int = -100;
+pub const AT_REMOVEDIR: ::c_int = 8;
 pub const F_DUPFD_CLOEXEC: ::c_int = 2;
 pub const F_GETFL: ::c_int = 5;
 pub const F_SETFL: ::c_int = 6;
@@ -318,7 +319,9 @@ pub const O_ACCMODE: ::c_int = 7;
 pub const O_APPEND: ::c_int = 8;
 pub const O_CLOEXEC: ::c_int = 0x4000;
 pub const O_CREAT: ::c_int = 0x10;
+pub const O_DIRECTORY: ::c_int = 0x00020;
 pub const O_EXCL: ::c_int = 0x40;
+pub const O_NOFOLLOW: ::c_int = 0x100;
 pub const O_NONBLOCK: ::c_int = 0x400;
 pub const O_RDONLY: ::c_int = 2;
 pub const O_RDWR: ::c_int = 3;
-- 
2.35.1

