From f6b9da9a07c8c6cae07c8893c064c6ce92a04175 Mon Sep 17 00:00:00 2001
From: no92 <no92.mail@gmail.com>
Date: Mon, 13 Nov 2023 02:55:47 +0100
Subject: [PATCH 7/7] managarm: fix termios ABI

---
 src/unix/mlibc/mod.rs | 54 +++++++++++++++++++++++++++++++++----------
 1 file changed, 42 insertions(+), 12 deletions(-)

diff --git a/src/unix/mlibc/mod.rs b/src/unix/mlibc/mod.rs
index 638dba0..8814b5e 100644
--- a/src/unix/mlibc/mod.rs
+++ b/src/unix/mlibc/mod.rs
@@ -220,20 +220,37 @@ pub const B4800: ::c_uint = 12;
 pub const B9600: ::c_uint = 13;
 pub const B19200: ::c_uint = 14;
 pub const B38400: ::c_uint = 15;
-pub const B57600: ::c_uint = 16;
-pub const B115200: ::c_uint = 17;
-pub const B230400: ::c_uint = 18;
+pub const B57600: ::c_uint = 0x1001;
+pub const B115200: ::c_uint = 0x1002;
+pub const B230400: ::c_uint = 0x1003;
+pub const B460800: ::c_uint = 0x1004;
+pub const B500000: ::c_uint = 0x1005;
+pub const B576000: ::c_uint = 0x1006;
+pub const B921600: ::c_uint = 0x1007;
+pub const B1000000: ::c_uint = 0x1008;
+pub const B1152000: ::c_uint = 0x1009;
+pub const B1500000: ::c_uint = 0x100A;
+pub const B2000000: ::c_uint = 0x100B;
+pub const B2500000: ::c_uint = 0x100C;
+pub const B3000000: ::c_uint = 0x100D;
+pub const B3500000: ::c_uint = 0x100E;
+pub const B4000000: ::c_uint = 0x100F;
 pub const BRKINT: ::c_uint = 0x0002;
 pub const BS0: ::c_int = 0x0000;
 pub const BS1: ::c_int = 0x2000;
 pub const BSDLY: ::c_int = 0x2000;
+pub const CBAUD: ::c_uint = 0x100F;
+pub const CBAUDEX: ::c_int =  0x1000;
+pub const CIBAUD: ::c_int =  0x100F0000;
 pub const CLOCAL: ::c_int = 0x0800;
+pub const CMSPAR: ::c_int =  0x40000000;
 pub const CR0: ::c_int = 0x0000;
 pub const CR1: ::c_int = 0x0200;
 pub const CR2: ::c_int = 0x0400;
 pub const CR3: ::c_int = 0x0600;
 pub const CRDLY: ::c_int = 0x0600;
 pub const CREAD: ::c_int = 0x0080;
+pub const CRTSCTS: ::c_int =  0x80000000;
 pub const CS5: ::c_uint = 0x0000;
 pub const CS6: ::c_int = 0x0010;
 pub const CS7: ::c_int = 0x0020;
@@ -241,13 +258,19 @@ pub const CS8: ::c_int = 0x0030;
 pub const CSIZE: ::c_int = 0x0030;
 pub const CSTOPB: ::c_int = 0x0040;
 pub const ECHO: ::c_uint = 0x0008;
+pub const ECHOCTL: ::c_int =  0x0200;
 pub const ECHOE: ::c_int = 0x0010;
 pub const ECHOK: ::c_int = 0x0020;
+pub const ECHOKE: ::c_int =  0x0800;
 pub const ECHONL: ::c_int = 0x0040;
 pub const ECHOPRT: ::c_int = 0x0400;
+pub const EXTA: ::c_int =  0x000E;
+pub const EXTB: ::c_int =  0x000F;
+pub const EXTPROC: ::c_int =  0x00010000;
 pub const FF0: ::c_int = 0x0000;
 pub const FF1: ::c_int = 0x8000;
 pub const FFDLY: ::c_int = 0x8000;
+pub const FLUSHO: ::c_int =  0x1000;
 pub const HUPCL: ::c_int = 0x0400;
 pub const ICANON: ::c_int = 0x0002;
 pub const ICRNL: ::c_uint = 0x0100;
@@ -255,10 +278,13 @@ pub const IEXTEN: ::c_int = 0x8000;
 pub const IGNBRK: ::c_uint = 0x0001;
 pub const IGNCR: ::c_uint = 0x0080;
 pub const IGNPAR: ::c_uint = 0x0004;
+pub const IMAXBEL: ::c_uint = 0x2000;
 pub const INLCR: ::c_uint = 0x0080;
 pub const INPCK: ::c_uint = 0x0010;
 pub const ISIG: ::c_int = 0x0001;
 pub const ISTRIP: ::c_uint = 0x0020;
+pub const IUCLC: ::c_uint = 0x0200;
+pub const IUTF8: ::c_uint = 0x4000;
 pub const IXANY: ::c_uint = 0x0800;
 pub const IXOFF: ::c_uint = 0x1000;
 pub const IXON: ::c_uint = 0x0400;
@@ -270,6 +296,7 @@ pub const NOFLSH: ::c_int = 0x0080;
 pub const OCRNL: ::c_int = 0x0008;
 pub const OFDEL: ::c_int = 0x0080;
 pub const OFILL: ::c_int = 0x0040;
+pub const OLCUC: ::c_int = 0x0002;
 pub const ONLCR: ::c_int = 0x0004;
 pub const ONLRET: ::c_int = 0x0020;
 pub const ONOCR: ::c_int = 0x0010;
@@ -277,24 +304,27 @@ pub const OPOST: ::c_uint = 0x0001;
 pub const PARENB: ::c_int = 0x0100;
 pub const PARMRK: ::c_uint = 0x0008;
 pub const PARODD: ::c_int = 0x0200;
+pub const PENDIN: ::c_int =  0x4000;
 pub const TAB0: ::c_int = 0x0000;
 pub const TAB1: ::c_int = 0x0800;
 pub const TAB2: ::c_int = 0x01000;
 pub const TAB3: ::c_int = 0x01800;
 pub const TABDLY: ::c_int = 0x01800;
-pub const TCIFLUSH: ::c_int = 1;
-pub const TCIOFF: ::c_int = 1;
+pub const TCIFLUSH: ::c_int = 0;
+pub const TCIOFF: ::c_int = 2;
 pub const TCIOFLUSH: ::c_int = 2;
-pub const TCION: ::c_int = 2;
-pub const TCOFLUSH: ::c_int = 3;
-pub const TCOOFF: ::c_int = 3;
-pub const TCOON: ::c_int = 4;
-pub const TCSADRAIN: ::c_int = 2;
-pub const TCSAFLUSH: ::c_int = 3;
-pub const TCSANOW: ::c_int = 1;
+pub const TCION: ::c_int = 3;
+pub const TCOFLUSH: ::c_int = 1;
+pub const TCOOFF: ::c_int = 0;
+pub const TCOON: ::c_int = 1;
+pub const TCSADRAIN: ::c_int = 1;
+pub const TCSAFLUSH: ::c_int = 2;
+pub const TCSANOW: ::c_int = 0;
 pub const TOSTOP: ::c_int = 0x0100;
 pub const VT0: ::c_int = 0x0000;
 pub const VT1: ::c_int = 0x4000;
+pub const XCASE: ::c_int = 0x0004;
+pub const XTABS: ::c_int =  0x1800;
 
 // The following are usize since they are indices into termios.c_cc
 pub const VINTR: usize = 0;
-- 
2.42.0

