From 858ee78090f83b5323e8665ed12316fe0f04f546 Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Mon, 29 Aug 2022 00:54:41 +0200
Subject: [PATCH] Remove calls to versioned binaries

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 Makefile.devel            | 4 ++--
 libcharset/Makefile.devel | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.devel b/Makefile.devel
index 3102ad2..d49e74a 100644
--- a/Makefile.devel
+++ b/Makefile.devel
@@ -6,8 +6,8 @@ SHELL = /bin/sh
 MAKE = make
 AUTOCONF = autoconf
 AUTOHEADER = autoheader
-AUTOMAKE = automake-1.16
-ACLOCAL = aclocal-1.16
+AUTOMAKE = automake
+ACLOCAL = aclocal
 GPERF = gperf
 CC = gcc -Wall
 CFLAGS = -O
diff --git a/libcharset/Makefile.devel b/libcharset/Makefile.devel
index 7a991cd..3e0e2ca 100644
--- a/libcharset/Makefile.devel
+++ b/libcharset/Makefile.devel
@@ -5,7 +5,7 @@ SHELL = /bin/sh
 MAKE = make
 AUTOCONF = autoconf
 AUTOHEADER = autoheader
-ACLOCAL = aclocal-1.16
+ACLOCAL = aclocal
 CP = cp
 RM = rm -f
 
-- 
2.37.2

