From e07ffa526bad1e7c8dd7d5197405ee57088056b8 Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Mon, 13 Apr 2026 13:28:17 +0200
Subject: [PATCH] Switch to a patched qtbase repo

---
 .gitmodules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.gitmodules b/.gitmodules
index a7e7ef1..ecdbb48 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,6 +1,6 @@
 [submodule "qtbase"]
 	path = qtbase
-	url = ../qtbase.git
+	url = https://github.com/Dennisbonke/qtbase.git
 	branch = 6.11.0
 	status = essential
 [submodule "qtsvg"]
-- 
2.53.0

