From ca8d35f53741f2043b3a75a597858696e6f07d6c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20S=C5=82omi=C5=84ski?=
 <kacper.slominski72@gmail.com>
Date: Wed, 6 Nov 2024 12:28:37 +0100
Subject: [PATCH] Use absolute URLs for submodules

---
 .gitmodules | 100 ++++++++++++++++++++++++++--------------------------
 1 file changed, 50 insertions(+), 50 deletions(-)

diff --git a/.gitmodules b/.gitmodules
index ee888bb..c3e6b44 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -6,47 +6,47 @@
 [submodule "qtsvg"]
 	depends = qtbase
 	path = qtsvg
-	url = ../qtsvg.git
+	url = https://code.qt.io/qt/qtsvg.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtdeclarative"]
 	depends = qtbase
 	recommends = qtimageformats qtshadertools qtsvg qtlanguageserver
 	path = qtdeclarative
-	url = ../qtdeclarative.git
+	url = https://code.qt.io/qt/qtdeclarative.git
 	branch = 6.6.2
 	status = essential
 [submodule "qtactiveqt"]
 	depends = qtbase
 	path = qtactiveqt
-	url = ../qtactiveqt.git
+	url = https://code.qt.io/qt/qtactiveqt.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtmultimedia"]
 	depends = qtbase qtshadertools
 	recommends = qtdeclarative qtquick3d
 	path = qtmultimedia
-	url = ../qtmultimedia.git
+	url = https://code.qt.io/qt/qtmultimedia.git
 	branch = 6.6.2
 	status = addon
 [submodule "qttools"]
 	depends = qtbase
 	recommends = qtdeclarative qtactiveqt
 	path = qttools
-	url = ../qttools.git
+	url = https://code.qt.io/qt/qttools.git
 	branch = 6.6.2
 	status = essential
 [submodule "qtxmlpatterns"]
 	depends = qtbase
 	recommends = qtdeclarative
 	path = qtxmlpatterns
-	url = ../qtxmlpatterns.git
+	url = https://code.qt.io/qt/qtxmlpatterns.git
 	branch = dev
 	status = ignore
 [submodule "qttranslations"]
 	depends = qttools
 	path = qttranslations
-	url = ../qttranslations.git
+	url = https://code.qt.io/qt/qttranslations.git
 	branch = 6.6.2
 	status = essential
 	priority = 30
@@ -54,20 +54,20 @@
 	depends = qtdeclarative qttools
 	recommends = qtmultimedia qtshadertools qtwebengine
 	path = qtdoc
-	url = ../qtdoc.git
+	url = https://code.qt.io/qt/qtdoc.git
 	branch = 6.6.2
 	status = essential
 	priority = 40
 [submodule "qtrepotools"]
 	path = qtrepotools
-	url = ../qtrepotools.git
+	url = https://code.qt.io/qt/qtrepotools.git
 	branch = master
 	status = essential
 	project = -
 [submodule "qtqa"]
 	depends = qtbase
 	path = qtqa
-	url = ../qtqa.git
+	url = https://code.qt.io/qt/qtqa.git
 	branch = dev
 	status = essential
 	priority = 50
@@ -75,264 +75,264 @@
 	depends = qtbase qtpositioning
 	recommends = qtdeclarative
 	path = qtlocation
-	url = ../qtlocation.git
+	url = https://code.qt.io/qt/qtlocation.git
 	branch = 6.6.2
 	status = preview
 [submodule "qtpositioning"]
 	depends = qtbase
 	recommends = qtdeclarative qtserialport
 	path = qtpositioning
-	url = ../qtpositioning.git
+	url = https://code.qt.io/qt/qtpositioning.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtsensors"]
 	depends = qtbase
 	recommends = qtdeclarative
 	path = qtsensors
-	url = ../qtsensors.git
+	url = https://code.qt.io/qt/qtsensors.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtsystems"]
 	depends = qtbase
 	recommends = qtdeclarative
 	path = qtsystems
-	url = ../qtsystems.git
+	url = https://code.qt.io/qt/qtsystems.git
 	branch = dev
 	status = ignore
 [submodule "qtfeedback"]
 	depends = qtdeclarative
 	recommends = qtmultimedia
 	path = qtfeedback
-	url = ../qtfeedback.git
+	url = https://code.qt.io/qt/qtfeedback.git
 	branch = master
 	status = ignore
 [submodule "qtpim"]
 	depends = qtdeclarative
 	path = qtpim
-	url = ../qtpim.git
+	url = https://code.qt.io/qt/qtpim.git
 	branch = dev
 	status = ignore
 [submodule "qtconnectivity"]
 	depends = qtbase
 	recommends = qtdeclarative
 	path = qtconnectivity
-	url = ../qtconnectivity.git
+	url = https://code.qt.io/qt/qtconnectivity.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtwayland"]
 	depends = qtbase
 	recommends = qtdeclarative
 	path = qtwayland
-	url = ../qtwayland.git
+	url = https://code.qt.io/qt/qtwayland.git
 	branch = 6.6.2
 	status = addon
 [submodule "qt3d"]
 	depends = qtbase
 	recommends = qtdeclarative qtshadertools qtmultimedia
 	path = qt3d
-	url = ../qt3d.git
+	url = https://code.qt.io/qt/qt3d.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtimageformats"]
 	depends = qtbase
 	path = qtimageformats
-	url = ../qtimageformats.git
+	url = https://code.qt.io/qt/qtimageformats.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtserialbus"]
 	depends = qtbase
 	recommends = qtserialport
 	path = qtserialbus
-	url = ../qtserialbus.git
+	url = https://code.qt.io/qt/qtserialbus.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtserialport"]
 	depends = qtbase
 	path = qtserialport
-	url = ../qtserialport.git
+	url = https://code.qt.io/qt/qtserialport.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtwebsockets"]
 	depends = qtbase
 	recommends = qtdeclarative
 	path = qtwebsockets
-	url = ../qtwebsockets.git
+	url = https://code.qt.io/qt/qtwebsockets.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtwebchannel"]
 	depends = qtbase
 	recommends = qtdeclarative qtwebsockets
 	path = qtwebchannel
-	url = ../qtwebchannel.git
+	url = https://code.qt.io/qt/qtwebchannel.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtwebengine"]
 	depends = qtdeclarative
 	recommends = qtwebchannel qttools qtpositioning
 	path = qtwebengine
-	url = ../qtwebengine.git
+	url = https://code.qt.io/qt/qtwebengine.git
 	branch = 6.6.2
 	status = addon
 	priority = 10
 [submodule "qtcanvas3d"]
 	depends = qtdeclarative
 	path = qtcanvas3d
-	url = ../qtcanvas3d.git
+	url = https://code.qt.io/qt/qtcanvas3d.git
 	branch = dev
 	status = ignore
 [submodule "qtwebview"]
 	depends = qtdeclarative
 	recommends = qtwebengine
 	path = qtwebview
-	url = ../qtwebview.git
+	url = https://code.qt.io/qt/qtwebview.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtcharts"]
 	depends = qtbase
 	recommends = qtdeclarative qtmultimedia
 	path = qtcharts
-	url = ../qtcharts.git
+	url = https://code.qt.io/qt/qtcharts.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtdatavis3d"]
 	depends = qtbase
 	recommends = qtdeclarative qtmultimedia
 	path = qtdatavis3d
-	url = ../qtdatavis3d.git
+	url = https://code.qt.io/qt/qtdatavis3d.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtvirtualkeyboard"]
 	depends = qtbase qtdeclarative qtsvg
 	recommends = qtmultimedia
 	path = qtvirtualkeyboard
-	url = ../qtvirtualkeyboard.git
+	url = https://code.qt.io/qt/qtvirtualkeyboard.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtgamepad"]
 	depends = qtbase
 	recommends = qtdeclarative
 	path = qtgamepad
-	url = ../qtgamepad.git
+	url = https://code.qt.io/qt/qtgamepad.git
 	branch = dev
 	status = ignore
 [submodule "qtscxml"]
 	depends = qtbase qtdeclarative
 	path = qtscxml
-	url = ../qtscxml.git
+	url = https://code.qt.io/qt/qtscxml.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtspeech"]
 	depends = qtbase
 	recommends = qtdeclarative qtmultimedia
 	path = qtspeech
-	url = ../qtspeech.git
+	url = https://code.qt.io/qt/qtspeech.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtnetworkauth"]
 	depends = qtbase
 	path = qtnetworkauth
-	url = ../qtnetworkauth.git
+	url = https://code.qt.io/qt/qtnetworkauth.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtremoteobjects"]
 	depends = qtbase
 	recommends = qtdeclarative
 	path = qtremoteobjects
-	url = ../qtremoteobjects.git
+	url = https://code.qt.io/qt/qtremoteobjects.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtwebglplugin"]
 	depends = qtbase qtwebsockets
 	recommends = qtdeclarative
 	path = qtwebglplugin
-	url = ../qtwebglplugin.git
+	url = https://code.qt.io/qt/qtwebglplugin.git
 	branch = dev
 	status = ignore
 [submodule "qtlottie"]
 	depends = qtbase qtdeclarative
 	path = qtlottie
-	url = ../qtlottie.git
+	url = https://code.qt.io/qt/qtlottie.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtquicktimeline"]
 	depends = qtbase qtdeclarative
 	path = qtquicktimeline
-	url = ../qtquicktimeline
+	url = https://code.qt.io/qt/qtquicktimeline
 	branch = 6.6.2
 	status = addon
 [submodule "qtquick3d"]
 	depends = qtbase qtdeclarative qtshadertools
 	recommends = qtquicktimeline
 	path = qtquick3d
-	url = ../qtquick3d.git
+	url = https://code.qt.io/qt/qtquick3d.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtshadertools"]
 	depends = qtbase
 	path = qtshadertools
-	url = ../qtshadertools.git
+	url = https://code.qt.io/qt/qtshadertools.git
 	branch = 6.6.2
 	status = addon
 [submodule "qt5compat"]
 	depends = qtbase qtdeclarative
 	path = qt5compat
-	url = ../qt5compat.git
+	url = https://code.qt.io/qt/qt5compat.git
 	branch = 6.6.2
 	status = deprecated
 [submodule "qtcoap"]
 	depends = qtbase
 	path = qtcoap
-	url = ../qtcoap.git
+	url = https://code.qt.io/qt/qtcoap.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtmqtt"]
 	depends = qtbase qtdeclarative
 	path = qtmqtt
-	url = ../qtmqtt.git
+	url = https://code.qt.io/qt/qtmqtt.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtopcua"]
 	depends = qtbase qtdeclarative
 	path = qtopcua
-	url = ../qtopcua.git
+	url = https://code.qt.io/qt/qtopcua.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtlanguageserver"]
 	depends = qtbase
 	path = qtlanguageserver
-	url = ../qtlanguageserver.git
+	url = https://code.qt.io/qt/qtlanguageserver.git
 	branch = 6.6.2
 	status = preview
 [submodule "qthttpserver"]
 	depends = qtbase
 	recommends = qtwebsockets
 	path = qthttpserver
-	url = ../qthttpserver.git
+	url = https://code.qt.io/qt/qthttpserver.git
 	branch = 6.6.2
 	status = preview
 [submodule "qtquick3dphysics"]
 	depends = qtbase qtdeclarative qtquick3d qtshadertools
 	path = qtquick3dphysics
-	url = ../qtquick3dphysics.git
+	url = https://code.qt.io/qt/qtquick3dphysics.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtgrpc"]
 	depends = qtbase
 	recommends = qtdeclarative
 	path = qtgrpc
-	url = ../qtgrpc.git
+	url = https://code.qt.io/qt/qtgrpc.git
 	branch = 6.6.2
 	status = preview
 [submodule "qtquickeffectmaker"]
 	depends = qtbase qtdeclarative qtshadertools
 	recommends = qtquick3d
 	path = qtquickeffectmaker
-	url = ../qtquickeffectmaker.git
+	url = https://code.qt.io/qt/qtquickeffectmaker.git
 	branch = 6.6.2
 	status = addon
 [submodule "qtgraphs"]
 	depends = qtbase qtdeclarative qtquick3d
 	path = qtgraphs
-	url = ../qtgraphs.git
+	url = https://code.qt.io/qt/qtgraphs.git
 	branch = 6.6.2
 	status = preview
-- 
2.45.2

