From f6606f7f836162222037efd9c93412b3c4e5032b Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Sat, 2 Mar 2024 12:46:53 +0100
Subject: [PATCH] Switch to a patched qtbase repo

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 .gitmodules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.gitmodules b/.gitmodules
index 1ad0f47..ee888bb 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,6 +1,6 @@
 [submodule "qtbase"]
 	path = qtbase
-	url = ../qtbase.git
+	url = https://github.com/Dennisbonke/qtbase.git
 	branch = 6.6.2
 	status = essential
 [submodule "qtsvg"]
-- 
2.43.0

