From ec27a48f63e0767d291704cd4c0ee3009b8b9730 Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Thu, 5 Mar 2026 21:54:20 +0100
Subject: [PATCH] Add Managarm support

---
 Configurations/10-main.conf | 61 +++++++++++++++++++++++++++++++++++++
 1 file changed, 61 insertions(+)

diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index 7e93968..43e93b6 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -1010,6 +1010,67 @@ my %targets = (
         ranlib           => "true",
     },
 
+#### Managarm
+    "managarm-generic" => {
+        inherit_from        => [ "BASE_unix" ],
+        CFLAGS          => picker(default   => "-Wall",
+                        debug       => "-O0 -g",
+                        release     => "-O3"),
+        CXXFLAGS        => picker(default   => "-Wall",
+                        debug       => "-O0 -g",
+                        release     => "-O3"),
+        cflags           => threads("-pthread"),
+        cxxflags         => combine("-std=c++11", threads("-pthread")),
+        lib_cppflags     => "-DOPENSSL_USE_NODELETE",
+        ex_libs          => add("-ldl", threads("-pthread")),
+        bn_ops           => "SIXTY_FOUR_BIT_LONG RC4_CHAR",
+        thread_scheme       => "pthreads",
+        dso_scheme      => "dlfcn",
+        shared_target       => "linux-shared",
+        shared_cflag        => "-fPIC",
+        shared_ldflag       => sub { $disabled{pinshared} ? () : "-Wl,-znodelete" },
+        shared_extension    => ".so.\$(SHLIB_VERSION_NUMBER)",
+    },
+
+    # This breaks the usual configuration naming convention but it's more
+    # convenient for us, since it matches the target triples
+
+    "x86_64-managarm" => {
+        inherit_from        => [ "managarm-generic" ],
+        asm_arch         => 'x86_64',
+        perlasm_scheme      => "elf",
+        lib_cppflags     => add("-DL_ENDIAN"),
+        bn_ops           => "SIXTY_FOUR_BIT_LONG",
+
+        # Configure doesn't want to play nice and passes "--cross-compile-suffix"
+        # on to the GCC invocation
+        CC          => "x86_64-managarm-gcc",
+        CXX         => "x86_64-managarm-g++",
+
+    },
+    "aarch64-managarm" => {
+        inherit_from        => [ "managarm-generic" ],
+        perlasm_scheme      => "linux64",
+        asm_arch         => 'aarch64',
+        lib_cppflags     => add("-DL_ENDIAN"),
+
+        # Configure doesn't want to play nice and passes "--cross-compile-suffix"
+        # on to the GCC invocation
+        CC          => "aarch64-managarm-gcc",
+        CXX         => "aarch64-managarm-g++",
+    },
+    "riscv64-managarm" => {
+        inherit_from        => [ "managarm-generic" ],
+        perlasm_scheme      => "linux64",
+        asm_arch         => 'riscv64',
+        lib_cppflags     => add("-DL_ENDIAN"),
+
+        # Configure doesn't want to play nice and passes "--cross-compile-suffix"
+        # on to the GCC invocation
+        CC          => "riscv64-managarm-gcc",
+        CXX         => "riscv64-managarm-g++",
+    },
+
 #### *BSD
     "BSD-generic32" => {
         # As for thread cflag. Idea is to maintain "collective" set of
-- 
2.53.0

