From d3811a0350dcc9d53d919b5c7b69b92173f1f48e Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Tue, 1 Jun 2021 22:03:10 +0200
Subject: [PATCH 2/2] Add Managarm targets to the configuration file

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 Configurations/10-main.conf | 43 +++++++++++++++++++++++++++++++++++++
 1 file changed, 43 insertions(+)

diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index 8dc3e858df..4b4f684578 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -911,6 +911,49 @@ my %targets = (
         ranlib           => "true",
     },
 
+#### Managarm
+    "managarm-generic64" => {
+        inherit_from     => [ "BASE_unix" ],
+        CFLAGS           => picker(default   => "-Wall",
+                         debug       => "-O0 -g",
+                         release     => "-O3"),
+        CXXFLAGS         => picker(default   => "-Wall",
+                         debug       => "-O0 -g",
+                         release     => "-O3"),
+        cxxflags         => add("-std=c++11"),
+        lib_cppflags     => combine("-DOPENSSL_USE_NODELETE", "-DL_ENDIAN"),
+        bn_ops           => "SIXTY_FOUR_BIT_LONG RC4_CHAR",
+        thread_scheme    => "pthreads",
+        dso_scheme       => "dlfcn",
+        shared_target    => "linux-shared",
+        shared_cflag     => "-fPIC",
+        shared_ldflag    => sub { $disabled{pinshared} ? () : "-Wl,-znodelete" },
+        shared_extension => ".so.\$(SHLIB_VERSION_NUMBER)",
+    },
+
+    # This breaks the usual configuration naming convention but it's more
+    # convenient for us, since it matches the target triples
+
+    "x86_64-managarm" => {
+        inherit_from     => [ "managarm-generic64", asm("x86_64_asm") ],
+        perlasm_scheme   => "elf",
+
+        # Configure doesn't want to play nice and passes "--cross-compile-suffix"
+        # on to the GCC invocation
+        CC               => "x86_64-managarm-gcc",
+        CXX              => "x86_64-managarm-g++",
+
+    },
+    "aarch64-managarm" => {
+        inherit_from     => [ "managarm-generic64" ],
+        perlasm_scheme   => "linux64",
+
+        # Configure doesn't want to play nice and passes "--cross-compile-suffix"
+        # on to the GCC invocation
+        CC               => "aarch64-managarm-gcc",
+        CXX              => "aarch64-managarm-g++",
+    },
+
 #### *BSD
     "BSD-generic32" => {
         # As for thread cflag. Idea is to maintain "collective" set of
-- 
2.31.1

