From 4168e727fa53063c0e7672461c371f12987deccd Mon Sep 17 00:00:00 2001
From: no92 <no92.mail@gmail.com>
Date: Tue, 7 Jun 2022 11:20:34 +0200
Subject: [PATCH] Add managarm-specific changes

---
 Src/zsh_system.h | 2 +-
 config.sub       | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Src/zsh_system.h b/Src/zsh_system.h
index 6f4efce..3b2cfd6 100644
--- a/Src/zsh_system.h
+++ b/Src/zsh_system.h
@@ -37,7 +37,7 @@
 #endif
 #endif
 
-#if defined(__linux) || defined(__GNU__) || defined(__GLIBC__) || defined(LIBC_MUSL) || defined(__CYGWIN__)
+#if defined(__linux) || defined(__GNU__) || defined(__GLIBC__) || defined(LIBC_MUSL) || defined(__CYGWIN__) || defined(__mlibc__)
 /*
  * Turn on numerous extensions.
  * This is in order to get the functions for manipulating /dev/ptmx.
diff --git a/config.sub b/config.sub
index 973a298..f5bce0f 100755
--- a/config.sub
+++ b/config.sub
@@ -135,7 +135,7 @@ case $1 in
 			| linux-newlib* | linux-musl* | linux-uclibc* | uclinux-uclibc* \
 			| uclinux-gnu* | kfreebsd*-gnu* | knetbsd*-gnu* | netbsd*-gnu* \
 			| netbsd*-eabi* | kopensolaris*-gnu* | cloudabi*-eabi* \
-			| storm-chaos* | os2-emx* | rtmk-nova*)
+			| storm-chaos* | os2-emx* | rtmk-nova* | managarm* )
 				basic_machine=$field1
 				os=$maybe_os
 				;;
@@ -1366,7 +1366,7 @@ case $os in
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
-	     | nsk* | powerunix* | genode*)
+	     | nsk* | powerunix* | genode* | managarm* )
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
-- 
2.51.1

