From 7f47e3293c11330c595cf855eaad617eb6d0a83d Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Sun, 5 Jan 2025 15:57:08 +0100
Subject: [PATCH] Add missing include

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 Src/jobs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Src/jobs.c b/Src/jobs.c
index a91ef78..023b038 100644
--- a/Src/jobs.c
+++ b/Src/jobs.c
@@ -29,6 +29,7 @@
 
 #include "zsh.mdh"
 #include "jobs.pro"
+#include <strings.h>
 
 /*
  * Job control in zsh
-- 
2.45.2

