From f5c1e0b546191c9b7aeea48067d2d5667ad2902c Mon Sep 17 00:00:00 2001
From: no92 <no92.mail@gmail.com>
Date: Mon, 31 Jul 2023 12:29:05 +0200
Subject: [PATCH] Patch out broken check

---
 meson.build | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/meson.build b/meson.build
index d7dd9b4..3ccbe81 100644
--- a/meson.build
+++ b/meson.build
@@ -363,23 +363,23 @@ if x11_dep.found() and xext_dep.found()
 
   # Can skip the run check by providing the result in a cross file or
   # native file as bool property value.
-  prop = meson.get_external_property('ipc_rmid_deferred_release', 'auto')
+  # prop = meson.get_external_property('ipc_rmid_deferred_release', 'auto')
   # We don't know the type of prop (bool, string) but need to differentiate
   # between a set value (bool) or the fallback value (string), so convert to
   # a string and chec the string value.
-  prop_str = '@0@'.format(prop)
-  if prop_str in ['true', 'false']
+  # prop_str = '@0@'.format(prop)
+  # if prop_str in ['true', 'false']
     ipc_rmid_deferred_release = (prop_str == 'true')
-    message('IPC_RMID_DEFERRED_RELEASE:', ipc_rmid_deferred_release)
-  elif prop_str == 'auto'
-    res = cc.run(files('meson-cc-tests/ipc_rmid_deferred_release.c'),
-      dependencies: [x11_dep, xext_dep],
-      name: 'shmctl IPC_RMID allowes subsequent attaches')
-
-    ipc_rmid_deferred_release = (res.returncode() == 0)
-  else
-    error('Unexpected value for external property ipc_rmid_deferred_release: @0@'.format(prop_str))
-  endif
+  #   message('IPC_RMID_DEFERRED_RELEASE:', ipc_rmid_deferred_release)
+  # elif prop_str == 'auto'
+  #   res = cc.run(files('meson-cc-tests/ipc_rmid_deferred_release.c'),
+  #     dependencies: [x11_dep, xext_dep],
+  #     name: 'shmctl IPC_RMID allowes subsequent attaches')
+
+  #   ipc_rmid_deferred_release = (res.returncode() == 0)
+  # else
+  #   error('Unexpected value for external property ipc_rmid_deferred_release: @0@'.format(prop_str))
+  # endif
 
   conf.set10('IPC_RMID_DEFERRED_RELEASE', ipc_rmid_deferred_release)
 endif
-- 
2.40.1

